### DISCUSSION TITLE
Improvements Schema Configuration Files

### DISCUSSION THREAD

**Discussion by ecarrenolozano** [biocypher/issues#5] [Ideas]
Hi dear BioCypher community,

It would be useful a kind of guide to know all the possible keywords allowed in the schema configuration file. This one should include descriptions in the context of Nodes and Edges.

For instance, by taking into account the [test schema config](https://github.com/biocypher/biocypher/blob/main/biocypher/_config/test_schema_config.yaml)  I could extract the following keywords:

- `db_collection_name`
- `exclude_properties`
- `inherit_properties`
- `input_label`
- `is_a`
- `label_as_edge`
- `preferred_id` 
- `properties`
- `represented_as`
- `source`
- `synonym_for`
- `target`

Q. In case of having a guide of this, where can  I find it?
Q. Are those all the possible keywords?

**Comment by slobentanzer** [comment]
@ecarrenolozano good idea, we should have that. Information about the keywords is fragmented across the entire documentation, it makes sense to keep them in a central place. Would you like to draft a docs page for this purpose?

`db_collection_name` is not a standard keyword, but it was introduced by the IGVF people that created a KG; this is the nature of the configuration: you can use other keywords for local functionalities without interfering with the default ones.

It does make a lot of sense to maintain an up-to-date doc specifically about them; I am happy to answer detailed questions after you have an initial draft. Could be a section in https://biocypher.org/installation.html#configuration

**Reply by ecarrenolozano** [reply]
Hi Sebastian,
I was thinking in something like [this](https://ecarrenolozano.github.io/mkdocs-material-biocypher/reference/schema-config-guide/#add-custom-fields). In there, the structure and the keywords with their descriptions are presented to the user. What do you think?

**Reply by slobentanzer** [reply]
Very nice, much better than what we have now!
